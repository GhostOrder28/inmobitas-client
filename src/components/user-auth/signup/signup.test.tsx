import * as redux from 'react-redux';
import Signup from './signup.component';
import { render, screen } from '@testing-library/react';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const store = {
  currentUser: {
    email: 'test@test.com',
    names: 'Test',
    userId: 1
  },
  errors: null
}


describe('Signup', () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;
  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector');
    spyOnUseSelector.mockReturnValue({
      email: 'test@test.com',
      names: 'Test',
      userId: 1
    });

    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);

  });


  it('renders Signup component', () => {
    expect(render(<Signup />)).toMatchSnapshot();
  })
})

