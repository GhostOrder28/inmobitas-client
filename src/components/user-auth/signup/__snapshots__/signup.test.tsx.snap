// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders Signup component 1`] = `
<div>
  <ReactFinalForm
    onSubmit={[Function]}
    render={[Function]}
  />
</div>
`;
